include/openmc/mesh.h

```diff
@@ -218,13 +241,178 @@ class RectilinearMesh : public Mesh
   bool intersects(Position& r0, Position r1, int* ijk) const;
 
   // Data members
-
   xt::xtensor<int, 1> shape_; //!< Number of mesh elements in each dimension
 
 private:
   std::vector<std::vector<double>> grid_;
 };
 
+#ifdef DAGMC
+
+class UnstructuredMesh : public Mesh {
+
+public:
+  UnstructuredMesh() = default;
+  UnstructuredMesh(pugi::xml_node);
+  ~UnstructuredMesh() = default;
+
+  void bins_crossed(const Particle* p,
+                    std::vector<int>& bins,
+                    std::vector<double>& lengths) const override;
+
+private:
```

Comment:
[NL.16](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rl-order): Use `public` before `protected` before `private` (there's currently a second public section below this)